### ISSUE TITLE
group_empty with custom label

### ISSUE THREAD

**Issue by JosefLitos** [nvim-tree/nvim-tree.lua#2515]
**Is your feature request related to a problem? Please describe.**
`group_empty` option allows for joining simple paths into one, resulting in a long path. The same can happen with `root_folder_label`, but there we can use a custom function that suits our needs and provides usually a shorter label. Such option would be very useful for grouped paths (for me at least).

**Describe the solution you'd like**
the simplest would be to allow `group_empty` option to be a function. Or maybe add a separate option in the likes of `grouped_folder_label` just for the function.

**Describe alternatives you've considered**
I tried looking into the code if there was an easy way to replace something, but I didn't get beyond [some form of recursion](https://github.com/nvim-tree/nvim-tree.lua/blob/c763861afb32f839555f9e797f4392f54ef4ad23/lua/nvim-tree/explorer/explore.lua#L73C22-L73C22). I didn't find a place where the names of the directories would get joined and rendered as a single point to inject my function.

**Additional context**
I shorten all paths to just their first letters of each word to fit into my little NvimTree window:
```lua
function(path)
	path = path:gsub(os.getenv 'HOME', '~', 1)
	return path:gsub('([a-zA-Z])[a-z]+', '%1') .. path:gsub('.*[^a-zA-Z].?', '', 1)
end
```

**Comment by Akmadan23** [comment]
This would be a really nice addition, I'll work on it and see if it's possible.

**Comment by Akmadan23** [comment]
> the simplest would be to allow group_empty option to be a function.

I implemented this version in #2521, can you please confirm if it works well for you?

**Comment by JosefLitos** [comment]
Yes, it works great. Thank you!
